core/pta/stats.c

```diff
@@ -137,6 +140,35 @@ static TEE_Result get_memleak_stats(uint32_t type,
 	return TEE_SUCCESS;
 }
 
+#if defined(CFG_TA_STATS)
+static TEE_Result get_user_ta_stats(uint32_t type,
+				    TEE_Param p[TEE_NUM_PARAMS])
+{
+	uint32_t res = TEE_SUCCESS;
+	/*
+	 * p[0].memref.buffer = output buffer to struct malloc_stats
+	 */
+	if (TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INOUT,
+			    TEE_PARAM_TYPE_MEMREF_OUTPUT,
+			    TEE_PARAM_TYPE_NONE,
+			    TEE_PARAM_TYPE_NONE) != type) {
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	if (p[0].value.a == STATS_INPUT_ZERO_SIZE)
+		res = tee_ta_instance_stats(NULL,
```

Comment:
You don't need this trick to get buffer size.
The command can pass a `buffer = NULL`  reference with a MEMREF_OUTPUT param and let TA update the `.size` field. Caller client see the expected size from output `.size` field.